### PR TITLE
gha: Auto assign fixup

### PR DESCRIPTION
**Pull request by jonkeane**
(1) be more robust to quoting issues, exploiting envvars
(2) continue-on-error for the debug step
(3) name the steps so it's a little bit clearer what's up 
(4) at least one comment explaining graphql

### CODE REVIEW

**Review by alistaire47 — APPROVED** [review]
Thanks!

File: .github/workflows/auto-assign-issues-prs.yml [hunk 1]
```diff
@@ -37,8 +41,15 @@ jobs:
           owner: ${{ github.repository_owner }}
           repo: ${{ github.event.repository.name }}
           pr_number: ${{ github.event.pull_request.number }}
-      - run: "echo 'Issue output: ${{ steps.get_linked_issue.outputs.data }}'"
-      - run: |
-          echo '${{ steps.get_linked_issue.outputs.data }}' | \
+      - name: Show the linked issue output
+        env: 
+          RETRIEVED_ISSUE: ${{ steps.get_linked_issue.outputs.data }}
+        run: "echo 'Issue output: $RETRIEVED_ISSUE'"
```
Comment (jgehrcke) [inline .github/workflows/auto-assign-issues-prs.yml]: Within single quotes the variable expansion is disabled :) (some of the things I try to remember about shells are about variable expansion :D).

**Review by jgehrcke — COMMENTED** [review]
Oh, and can we prefix the PR title with something like  `gha: ` or `ci: `? So that we can rather quickly see from the commit msg (after merge) which area this commit was touching.

**Review by austin3dickey — APPROVED** [review]

**Comment by alistaire47** [comment]
Unfortunately I'm not sure this worked; I just rebased and got an email the run failed (though it doesn't show in the build section?): https://github.com/conbench/conbench/actions/runs/4107714305/workflow

**Pull request merged**